sdks/python/apache_beam/runners/interactive/pipeline_instrument.py

```diff
@@ -375,12 +377,15 @@ def _read_cache(self, pipeline, pcoll):
     # Can only read from cache when the cache with expected key exists.
     if self._cache_manager.exists('full', key):
       if key not in self._cached_pcoll_read:
+        if pcoll.is_bounded:
+          read_transform = cache.ReadCache(self._cache_manager, key)
+        else:
+          read_transform = streaming_cache.StreamingCacheReceiver()
+
         # Mutates the pipeline with cache read transform attached
         # to root of the pipeline.
-        pcoll_from_cache = (
-            pipeline
-            | '{}{}'.format(READ_CACHE, key) >> cache.ReadCache(
-                self._cache_manager, key))
+        pcoll_from_cache = (pipeline
```

Comment:
Have these source generated pcolls been added to the `self.cacheables` so that they can get a `self.cache_key()`?
Or do you want to add a different execution path for these pcolls so that they can get unique cache keys?

Reply:
The cache keys should be the key of their output PCollection. I guess that means if it has multiple outputs it will have multiple read from cachces?